network/manager.go

```diff
@@ -382,11 +442,67 @@ func (nm *networkManager) DeleteEndpoint(networkID, endpointID string) error {
 	return nil
 }
 
+func (nm *networkManager) DeleteEndpointState(networkID string, epInfo *EndpointInfo) error {
+	nw := &network{
+		Id:           networkID,
+		Mode:         opModeTransparentVlan,
+		SnatBridgeIP: "",
+		extIf: &externalInterface{
+			Name:       InfraInterfaceName,
+			MacAddress: nil,
+		},
+	}
+
+	ep := &endpoint{
+		Id:                       epInfo.Id,
+		HnsId:                    epInfo.HNSEndpointID,
+		HostIfName:               epInfo.IfName,
+		LocalIP:                  "",
+		VlanID:                   0,
+		AllowInboundFromHostToNC: false,
+		AllowInboundFromNCToHost: false,
+		EnableSnatOnHost:         false,
+		EnableMultitenancy:       false,
+		NetworkContainerID:       epInfo.Id,
+	}
+	logger.Info("Deleting endpoint with", zap.String("Endpoint Info: ", epInfo.PrettyString()), zap.String("HNISID : ", ep.HnsId))
+	return nw.deleteEndpointImpl(netlink.NewNetlink(), platform.NewExecClient(logger), nil, nil, ep)
```

Comment:
we'd need to pass endpoint with SecondaryInterfaceInfo for deletion as well

Reply:
This is only going to cover stateless CNI for single Tenancy

Reply:
We are not covering Swift 2.0 at this point

Reply:
if we address swift 2.0, then we dont need to care about migration for swift2.0 later.. Not in this PR, probably a different PR